temporal-sdk/src/test/java/io/temporal/worker/StickyWorkerTest.java

```diff
@@ -471,74 +462,94 @@ public void workflowsCanBeQueriedAfterEviction() throws Exception {
 
     // Assert
     assertEquals(workflow.getProgress(), GreetingSignalWorkflow.Status.WAITING_FOR_NAME);
+    SDKTestWorkflowRule.assertNoHistoryEvent(
+        wrapper.testEnv.getWorkflowExecutionHistory(execution).getHistory(),
+        EventType.EVENT_TYPE_WORKFLOW_TASK_FAILED);
 
+    wrapper.close();
+  }
+
+  @Test
+  public void workflowCanContinueExecutionAfterBeingEvictedAndQueried() throws Exception {
+    // Arrange
+    String taskQueueName = "continueExecutionAfterEvictionAndQueryStickyTest";
+    TestEnvironmentWrapper wrapper =
+        new TestEnvironmentWrapper(WorkerFactoryOptions.newBuilder().build());
+    WorkerFactory factory = wrapper.getWorkerFactory();
+    Worker worker = factory.newWorker(taskQueueName);
+    worker.registerWorkflowImplementationTypes(GreetingSignalWorkflowImpl.class);
+    factory.start();
+
+    WorkflowOptions workflowOptions =
+        WorkflowOptions.newBuilder()
+            .setTaskQueue(taskQueueName)
+            .setWorkflowRunTimeout(Duration.ofDays(30))
+            .setWorkflowTaskTimeout(Duration.ofSeconds(30))
+            .setWorkflowIdReusePolicy(
+                WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE)
+            .build();
+    GreetingSignalWorkflow workflow =
+        wrapper.getWorkflowClient().newWorkflowStub(GreetingSignalWorkflow.class, workflowOptions);
+
+    // Act
+    WorkflowExecution execution = WorkflowClient.start(workflow::getGreeting);
+
+    Thread.sleep(200); // Wait for workflow to start
```

Comment:
In order to avoid flakiness, in Go SDKs and others we often have a query setup on the workflow and poll until it returns instead of relying on time.

Reply:
Yeah, same in Java, this class needs a little bit more love that I'm not ready to give now to do that I believe. It's a part of Test Rule in Java and this class doesn't use the test rule. But let me double check if I can hook it in anyway.